app/src/main/java/com/google/android/sambadocumentsprovider/provider/SambaDocumentsProvider.java

```diff
@@ -359,11 +429,46 @@ private Cursor buildErrorCursor(String[] projection, @StringRes int resId) {
         case Document.COLUMN_LAST_MODIFIED:
           row[i] = metadata.getLastModified();
           break;
+        case Document.COLUMN_ICON:
+          row[i] = metadata.getIconResourceId();
```

Comment:
If you're not providing icon, just pass -1, or remove this column from projection.

I don't think we want to provide the icon for regular folders.